executor/batch_checker.go

```diff
@@ -0,0 +1,252 @@
+// Copyright 2018 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package executor
+
+import (
+	"github.com/juju/errors"
+	"github.com/pingcap/tidb/kv"
+	"github.com/pingcap/tidb/sessionctx"
+	"github.com/pingcap/tidb/table"
+	"github.com/pingcap/tidb/table/tables"
+	"github.com/pingcap/tidb/tablecodec"
+	"github.com/pingcap/tidb/types"
+)
+
+type keyValue struct {
+	key   kv.Key
+	value []byte
+}
+
+type keyValueWithDupInfo struct {
+	newKeyValue keyValue
+	oldKeyValue keyValue
+	dupErr      error
+}
+
+type toBeCheckRow struct {
```

Comment:
s/toBeCheckRow/toBeCheckedRow

Reply:
Actually, I want a more meaningful name. Any suggestions?

Reply:
I think `toBeCheckedRow` is ok.